index.js

```diff
@@ -0,0 +1,166 @@
+const data = [
+    { "id": 1, "name": "Alan Smith", "age": 57, "country": "USA" },
+    { "id": 2, "name": "Nina Brown", "age": 32, "country": "Germany" },
+    { "id": 3, "name": "Kevin Sallivan", "age": 21, "country": "Canada" },
+    { "id": 4, "name": "Sergey Petrov", "age": 24, "country": "Russia" },
+    { "id": 5, "name": "Mina Leen", "age": 40, "country": "China" },
+    { "id": 6, "name": "Sam White", "age": 26, "country": "USA" },
+    { "id": 7, "name": "Peter Olsten", "age": 40, "country": "France" },
+    { "id": 8, "name": "Lina Rein", "age": 30, "country": "Germany" },
+    { "id": 9, "name": "Many Cute", "age": 22, "country": "Canada" },
+    { "id": 10, "name": "Andrew Wein", "age": 27, "country": "Italy" },
+    { "id": 11, "name": "Paolo Sanders", "age": 40, "country": "Spain" },
+    { "id": 12, "name": "Tanya Krieg", "age": 28, "country": "Germany" }
+]
+
+function changeBtnState(state, addClass, removeClass, id) {
+    const btn = $$(id);
+    btn.config.state = state;
+    const btnNode = btn.getNode();
+    webix.html.addCss(btnNode, `button_${addClass}`);
+    webix.html.removeCss(btnNode, `button_${removeClass}`);
+    btn.blur();
+    return btn.setValue(btn.config.states[btn.config.state]);
+}
+
+function sortDataInTable(key, dir, type) {
+    return $$("table").sort(key, dir, type);
+}
+
+webix.protoUI({
+    name: 'buttonStates',
+    $init(config) {
+        config.value = config.states[config.state];
+        this.$view.className += " button_off";
+        this.attachEvent("onItemClick", () => {
+            switch (this.config.state) {
+                case 0:
+                    changeBtnState(1, "sort_asc", "off", this.config.id);
+                    break;
+                case 1:
+                    changeBtnState(2, "sort_desc", "sort_asc", this.config.id);
+                    break;
+                case 2:
+                    changeBtnState(0, "off", "sort_desc", this.config.id);
+                    break;
+            }
+        })
+    }
+}, webix.ui.button)
+
+const tabs = {
+    view: "tabbar",
+    multiview: true,
+    options: [
+        { id: "first_task", value: "First task" },
+        { id: "second_task", value: "Second task" },
+    ]
+}
+
+const firstTask = {
+    rows: [
+        {
+            cols: [{
+                view: "label",
+                label: "Sort list",
+                autowidth: true
+            },
+            {
+                view: "buttonStates",
+                width: 270,
+                states: { 0: "Off", 1: "Sort Asc", 2: "Sort Desc" },
+                state: 0,
+                on: {
+                    onItemClick() {
```

Comment:
Не совсем по заданию
> ...fire the state change event, so that we can sort/unsort our data correctly. 

Логика связана с onItemClick, но напрямую он не используется
- по клику меняется state
- после смены state вызывается кастомное событие, куда аргументом попадает новый state
- и именно на него здесь нужно подписаться, а не на сам onItemClick